public/app/features/datasources/components/picker/DataSourceDropdown.tsx

```diff
@@ -24,21 +23,30 @@ export function DataSourceDropdown(props: DataSourceDropdownProps) {
   const [isOpen, setOpen] = useState(false);
   const [markerElement, setMarkerElement] = useState<HTMLInputElement | null>();
   const [selectorElement, setSelectorElement] = useState<HTMLDivElement | null>();
-  const [filterTerm, setFilterTerm] = useState<string>();
 
   const currentDataSourceInstanceSettings = useDatasource(current);
+  const [filterTerm, setFilterTerm] = useState<string>(dataSourceLabel(currentDataSourceInstanceSettings));
 
   const popper = usePopper(markerElement, selectorElement, {
     placement: 'bottom-start',
   });
 
+  const onClose = useCallback(() => {
+    setFilterTerm(dataSourceLabel(currentDataSourceInstanceSettings));
+    setOpen(false);
+    markerElement?.blur();
+  }, [setOpen, markerElement, setFilterTerm, currentDataSourceInstanceSettings]);
+
+  const onOpen = useCallback(() => {
```

Comment:
I don't see the need to use `useCallback` here since we don't rely on this prop to rerender a component. Am I right?